pkg/media/h264reader/h264reader.go

```diff
@@ -0,0 +1,272 @@
+				h.buf = append(h.buf, data[*lastPrefixOffset:]...)
+			} else {
+				// a prefix was not found, save all data into the temporary buffer
+				h.buf = append(h.buf, data...)
+			}
+			break
+		}
+		b := data[i]
+		i++
+		switch b {
+		case 0x00:
+			{
+				if h.LastNullCount < 3 {
+					h.LastNullCount++
+				}
+				continue
+			}
+		case 0x01:
+			{
+				if h.LastNullCount >= 2 { // found a NAL prefix 0x00_00_01 or 0x00_00_00_01
+
+					prefixOffset := i
+					if lastPrefixOffset != nil {
+						// NAL is a part of data from the end of the last prefix to the beginning of the current prefix. Save it
+						size := (i - h.LastNullCount) - *lastPrefixOffset - 1
+						if size > 0 && h.PrefixCount > 0 {
+							nal := data[*lastPrefixOffset : *lastPrefixOffset+size]
+							// save nal
+							nals = append(nals, nal)
+						}
+					} else {
+						// a previous (last) prefix isn't exist
+						// NAL is the temporary buffer and a part of data from the beginning to the current prefix
+						size := i - h.LastNullCount - 1
+						nal := make([]byte, 0)
+						if size < 0 {
+							if len(h.buf) > 0 {
+								nal = append(nal, h.buf[0:len(h.buf)+size]...)
+							}
+						} else {
+							nal = append(nal, h.buf...)
+							nal = append(nal, data[0:size]...)
+						}
+
+						// save non-empty NAL only after at least one prefix was detected
+						if len(nal) > 0 && h.PrefixCount > 0 {
+							nals = append(nals, nal)
+						}
+						h.buf = make([]byte, 0)
+					}
+					p := prefixOffset
+					lastPrefixOffset = &p
+					h.PrefixCount++
+				}
+			}
+		default:
+		}
+		h.LastNullCount = 0
+	}
+	return nals
+}
+
+type nalUnitType uint8
+
+const ( //   Table 7-1 NAL unit type codes
+	unspecified              nalUnitType = 0  // unspecified
+	codedSliceNonIdr         nalUnitType = 1  // Coded slice of a non-IDR picture
+	codedSliceDataPartitionA nalUnitType = 2  // Coded slice data partition A
+	codedSliceDataPartitionB nalUnitType = 3  // Coded slice data partition B
+	codedSliceDataPartitionC nalUnitType = 4  // Coded slice data partition C
+	codedSliceIdr            nalUnitType = 5  // Coded slice of an IDR picture
+	sei                      nalUnitType = 6  // Supplemental enhancement information (sei)
+	sps                      nalUnitType = 7  // Sequence parameter set
+	pps                      nalUnitType = 8  // Picture parameter set
+	aud                      nalUnitType = 9  // Access unit delimiter
+	endOfSequence            nalUnitType = 10 // End of sequence
+	endOfStream              nalUnitType = 11 // End of stream
+	filler                   nalUnitType = 12 // filler data
+	spsExt                   nalUnitType = 13 // Sequence parameter set extension
+	// 14..18           // Reserved
+	nalUnitTypeCodedSliceAux nalUnitType = 19 // Coded slice of an auxiliary coded picture without partitioning
+	// 20..23           // Reserved
+	// 24..31           // unspecified
+)
+
+func nalUnitTypeStr(v nalUnitType) string {
```

Comment:
Mind changing this to be

```
func (n *nalUnitType) String() string { }
```

Reply:
I think about removing this function. It is not used in parser.

Reply:
done